addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -172,38 +176,62 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+                    logger.warn("Received unknown command {} for channel {}", command.toString(), CHANNEL_SWITCH);
                 }
                 break;
             case CHANNEL_SETTEMP:
                 if (command instanceof DecimalType) {
-                    BigDecimal temperature = new BigDecimal(command.toString());
+                    final BigDecimal temperature = new BigDecimal(command.toString());
+                    state.getHkr().setTist(temperature);
                     fritzBox.setSetTemp(ain, HeatingModel.fromCelsius(temperature));
+                    updateState(CHANNEL_RADIATOR_MODE, new StringType(state.getHkr().getRadiatorMode()));
+                } else if (command instanceof IncreaseDecreaseType) {
+                    final BigDecimal temperature = state.getHkr().getTsoll();
+                    if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                        temperature.add(BigDecimal.ONE);
+                    } else {
+                        temperature.subtract(BigDecimal.ONE);
+                    }
+                    state.getHkr().setTist(temperature);
+                    fritzBox.setSetTemp(ain, temperature);
+                    updateState(CHANNEL_RADIATOR_MODE, new StringType(state.getHkr().getRadiatorMode()));
                 } else if (command instanceof OnOffType) {
-                    BigDecimal temperature = command.equals(OnOffType.ON) ? HeatingModel.TEMP_FRITZ_ON
+                    final BigDecimal temperature = OnOffType.ON.equals(command) ? HeatingModel.TEMP_FRITZ_ON
                             : HeatingModel.TEMP_FRITZ_OFF;
+                    state.getHkr().setTist(temperature);
                     fritzBox.setSetTemp(ain, temperature);
+                    updateState(CHANNEL_RADIATOR_MODE, new StringType(state.getHkr().getRadiatorMode()));
+                } else {
+                    logger.warn("Received unknown command {} for channel {}", command.toString(), CHANNEL_SETTEMP);
                 }
                 break;
             case CHANNEL_RADIATOR_MODE:
                 if (command instanceof StringType) {
                     if (command.equals(MODE_ON)) {
-                        BigDecimal settemp = (BigDecimal) getThing().getConfiguration().get(THING_SETTEMP);
-                        fritzBox.setSetTemp(ain, HeatingModel.fromCelsius(settemp));
-                        updateState(CHANNEL_SETTEMP, new DecimalType(settemp));
+                        state.getHkr().setTist(HeatingModel.TEMP_FRITZ_ON);
+                        fritzBox.setSetTemp(ain, HeatingModel.TEMP_FRITZ_ON);
+                        updateState(CHANNEL_SETTEMP,
+                                new DecimalType(HeatingModel.toCelsius(HeatingModel.TEMP_FRITZ_ON)));
                     } else if (command.equals(MODE_OFF)) {
+                        state.getHkr().setTist(HeatingModel.TEMP_FRITZ_OFF);
                         fritzBox.setSetTemp(ain, HeatingModel.TEMP_FRITZ_OFF);
```

Comment:
*Design*: Now you have to combine every call to `fritzBox` with an update / call to the state object. Would it maybe make more sense to either wrap the `fritzBox` object with a class which only has this goal or maybe make it part of the logic of  `fritzBox`. Wdyt?